### PR TITLE
modify process to process in a series

### PR DESCRIPTION
**Pull request by anupvarghese**

### CODE REVIEW

File: src/runner/sqs.js [hunk 1]
```diff
@@ -84,37 +84,32 @@ class SqsRunner extends BaseRunner implements IRunner {
...
+    }
   };
 
-  process(topics: Array<string>) {
+  async process(topics: Array<string>) {
     const subscriptions = this.getActiveSubsciptions(topics);
-    this.logger.info('initializing consumer', subscriptions);
-    return Promise.all(subscriptions.map(async (topic) => {
-      const queueURL = await getUrl(this.sqs, topic);
+    for (const topic of subscriptions) { //eslint-disable-line
```
Comment (danielmoi) [inline src/runner/sqs.js]: specify the lint errors?

File: src/runner/redis.js [hunk 2]
```diff
@@ -68,29 +68,29 @@ class RedisRunner extends BaseRunner implements IRunner {
...
-      this.iterateOnQueue(topic);
-    }, this.config.consumerPollInterval);
-  };
+    }
+  }
 
-  process(topics: Array<string>) {
+  async process(topics: Array<string>) {
     const subscriptions = this.getActiveSubsciptions(topics);
     this.logger.info('initializing consumer', subscriptions);
```
Comment (anupvarghese) [inline src/runner/redis.js]: can we remove this logging, it will log every second.

File: src/runner/sqs.js [hunk 3]
```diff
@@ -82,39 +82,52 @@ class SqsRunner extends BaseRunner implements IRunner {
...
+  async getQueueUrls(subscriptions: Array<string>) {
+    if (Object.keys(this.sqsUrls).length === subscriptions.length) {
+      return;
+    }
+
+    const urls = await Promise.all(subscriptions.map(topic => (
+      getUrl(this.sqs, topic).then(url => ([topic, url]))
+    )));
+
+    for (const [topic, url] of urls) {
```
Comment (anupvarghese) [inline src/runner/sqs.js]: `no-restricted-syntax`

File: src/runner/sqs.js [hunk 4]
```diff
@@ -82,39 +82,52 @@ class SqsRunner extends BaseRunner implements IRunner {
...
-      return this.iterateOnQueue(params, topic);
-    }));
+
+      await this.dequeue(topic, params); // eslint-disable-line
+    }
+
+    setTimeout(this.process.bind(this), this.config.consumerPollInterval);
+  }
+
+  async getQueueUrls(subscriptions: Array<string>) {
```
Comment (anupvarghese) [inline src/runner/sqs.js]: Thanks for this :)

**Pull request merged**